lib/puppet/configurer.rb

```diff
@@ -402,7 +409,11 @@ def run_internal(options)
     end
 
     report.cached_catalog_status ||= @cached_catalog_status
-    report.add_times(:total, Time.now - start)
+    if options[:start_time]
+      report.add_times(:total, Time.now - options[:start_time])
+    else
+      report.add_times(:total, Time.now - report.time)
```

Comment:
on line 245 you're setting `report.time = options[:start_time]`. We shouldn't need this block at all, since `report.time` and `options[:start_time]` should be the same thing at this point in the code.

Reply:
Good catch, I wrote this part before modifying `report.time` and kind of forgot about it.